mobile/src/screens/ChatScreen.tsx

```diff
@@ -49,122 +58,294 @@ export const ChatScreen: React.FC = () => {
     }
   }, [roomId]);
 
-  // Chat room data changed (ie. chat was resolved / rated)
+  // Chat room data changed
   useEffect(() => {
     if (chatRoomData) {
-      const { id, resolved, rating } = chatRoomData;
-      if (resolved && rating) {
-        // Reset chat
-        AsyncStorage.removeItem('chatRoomId');
-        setMessages([]);
-        setRoomId(null);
-      } else if (messages.length === 0) {
+      if (messages.length === 0) {
         // Subscribe to new chat messages
-        return chatAPI.listenForNewMessages(id, onMessagesChange);
+        return chatAPI.listenForNewMessages(chatRoomData.id, onMessagesChange);
       }
     }
   }, [chatRoomData]);
 
   const sendMessage = async (): Promise<void> => {
     let newRoomId = roomId;
-    if (!newRoomId && auth.user) {
-      // Create a new room if no previous roomId
-      newRoomId = await chatAPI.createRoom(auth.user);
-      setRoomId(newRoomId);
-      AsyncStorage.setItem("chatRoomId", newRoomId);
-    }
-    if (newRoomId && auth.user) {
-      chatAPI.sendMessage(newRoomId, messageText, auth.user?.name);
-      setMessageText("");
+    if (auth.user) {
+      // Create a new room if no previous roomId OR if current chat is resolved
+      if (!newRoomId || (chatRoomData && chatRoomData.resolved)) {
+        setMessages([]);
+        newRoomId = await chatAPI.createRoom(auth.user);
+        setRoomId(newRoomId);
+        AsyncStorage.setItem('chatRoomId', newRoomId);
```

Comment:
Hmm we might actually need to scope this to the current user, so this doesn't happen:

1. user1 signs in
2. user1 creates chat room
3. user1 signs in, user2 signs out
4. user2 opens chat, cached room ID is not accessible to them

definitely an edge case and I don't think many families will use multiple accounts but it's an easy fix, just append the user ID to the asyncstorage key

Reply:
Oh yeah, nice catch. That would definitely be an issue!